test/integration/instance_firewalls_test.go

```diff
@@ -0,0 +1,49 @@
+package integration
+
+import (
+	"context"
+	"testing"
+
+	"github.com/linode/linodego"
+)
+
+func TestInstanceFirewalls_List(t *testing.T) {
+	client, instance, _, teardown, err := setupInstanceFirewall(t, []firewallModifier{
+		func(createOpts *linodego.FirewallCreateOptions) {
+			createOpts.Label = "linodego-fw-ins-test"
+		},
+	}, "fixtures/TestInstanceFirewalls_List")
+	if err != nil {
+		t.Error(err)
+	}
+	defer teardown()
+
+	result, err := client.ListInstanceFirewalls(context.Background(), instance.ID, nil)
+	if err != nil {
+		t.Errorf("Error listing Firewalls, expected struct, got error %v", err)
+	}
+
+	if len(result) == 0 {
+		t.Errorf("Expected a list of Firewalls, but got none: %v", err)
+	}
+}
+
+func setupInstanceFirewall(t *testing.T, firewallModifiers []firewallModifier, fixturesYaml string) (*linodego.Client, *linodego.Instance, *linodego.Firewall, func(), error) {
+	t.Helper()
+	client, instance, instanceTeardown, err := setupInstance(t, fixturesYaml,
+		func(client *linodego.Client, opts *linodego.InstanceCreateOptions) {
+			opts.Label = "linodego-fw-inst-test"
+		})
+	device := linodego.DevicesCreationOptions{Linodes: []int{instance.ID}}
+	firewallModifiers = append(firewallModifiers,
+		func(createOpts *linodego.FirewallCreateOptions) {
+			createOpts.Devices = device
+		})
+	firewall, firewallTeardown, err := createFirewall(t, client, firewallModifiers...)
+
+	teardown := func() {
+		instanceTeardown()
+		firewallTeardown()
```

Comment:
Due to some fixtures weirdness `setupInstance`'s teardown will [close the fixtures file](https://github.com/linode/linodego/blob/main/test/integration/integration_suite_test.go#L154C18-L154C18), so you'll need to run it after `firewallTeardown()` so the firewall DELETE request is stored in the fixture file. Alternatively, you could directly call [createTestClient](https://github.com/linode/linodego/blob/main/test/integration/integration_suite_test.go#L120) here and use [createInstance](https://github.com/linode/linodego/blob/main/test/integration/instances_test.go#L446C6-L446C20) rather than `setupInstance`.

This should fix the CI failures after re-running the fixtures :+1:

Reply:
Thank you for the explanation.